### PR TITLE
CYS: Disable readonly mode only when full composability feature flag is enabled

### PR DESCRIPTION
**Pull request by gigitux**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Head over to WooCommerce > Home > Customize your store.
2. Click on `Start Designing`.
3. Click on Design Your Homepage.
4. Click on blocks on the Assembler
5. Ensure that they are not highlighted.
6. Click on a pattern.
7. Ensure that any block isn't highlighted on the site preview.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

CYS: Disable readonly mode only when full composability feature flag is enabled.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/auto-block-preview.tsx [hunk 1]
```diff
@@ -241,7 +242,11 @@ function ScaledBlockPreview( {
 						// @ts-ignore disabled prop exists
 						scrolling={ isScrollable ? 'yes' : 'no' }
 						tabIndex={ -1 }
-						readonly={ false }
+						readonly={
+							isFullComposabilityFeatureAndAPIAvailable()
+								? false
+								: true
+						}
```
Comment (imanish003) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/auto-block-preview.tsx]: Nitpick: Maybe we can simplify this to following
```suggestion
						readonly={
							! isFullComposabilityFeatureAndAPIAvailable()
						}
```

**Review by imanish003 — APPROVED** [review]
I have tested the changes and it's working as expected. I am approving this PR. I left one minor nitpick. Feel free to merge it after addressing it. 🚀

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/auto-block-preview.tsx (hunk 1 above)
Comment (gigitux) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/auto-block-preview.tsx]: Fixed with [20d2870](https://github.com/woocommerce/woocommerce/pull/48752/commits/20d28707a7a8f3b15d71c26d17a5973f8033881e)

**Comment by gigitux** [comment]
I added another commit to ensure that the block is not selected when the user selects another pattern: [0f66ded](https://github.com/woocommerce/woocommerce/pull/48752/commits/0f66ded5c3dc79af4a76aa5237d37f457c1038dd)

**Comment by rodelgc** [comment]
Temporarily restoring the branch to rebase it to `release/9.1`. Will re-delete `fix/read-only-assembler` afterwards.

**Comment by rodelgc** [comment]
Conflicts were encountered when trying to cherry-pick this PR as a CFE into `release/9.1`. I created https://github.com/woocommerce/woocommerce/pull/49002 to resolve those conflicts and I merged that PR into the release branch instead of this one.

**Pull request merged**